File: split/newglue.pas
function newglue(q: halfword): halfword;
var
  p: halfword;
begin
  p := getnode(2);
  mem[p].hh.b0 := 10;
  mem[p].hh.b1 := 0;
  mem[p + 1].hh.rh := 0;
  mem[p + 1].hh.lh := q;
  mem[q].hh.rh := mem[q].hh.rh + 1;
  newglue := p;
end;

File: cpp/newglue.cpp
#include "newglue.h"
#include "getnode.h"

halfword newglue(halfword q)
{
	auto p = getnode(2);
	type(p) = glue_node;
	subtype(p) = 0;
	link(p+1) = 0;
	info(p+1) = q;
	link(q)++;
	return p;
}
